open_instruct/finetune.py

```diff
@@ -710,11 +699,33 @@ def main(args: FlatArguments):
                 clean_last_n_checkpoints(args.output_dir, args.keep_last_n_checkpoints)
             accelerator.wait_for_everyone()
 
-    # dont save state, just model to the top-level dir.
     if args.output_dir is not None:
         if accelerator.is_main_process:
             tokenizer.save_pretrained(args.output_dir)
-        save_with_accelerate(accelerator, model, tokenizer, args.output_dir, args)
+        save_with_accelerate(
+            accelerator,
+            model,
+            tokenizer,
+            args.output_dir,
+            args.use_lora,
+        )
+
+    if accelerator.is_main_process:
+        if is_beaker_job() and args.try_launch_beaker_eval_jobs and args.output_dir in ["/output/", "/output"]:
+            submit_beaker_eval_jobs(model_name=run_name, location=beaker_config.beaker_dataset_ids[0])
+            args.try_launch_beaker_eval_jobs = False # don't launch the eval jobs for HF model then
```

Comment:
FYI this could be the code to support launching eval using beaker dataset. However, I just realized the beaker dataset might not get uploaded until the job is finished, so I will be removing this code and focus mainly on eval for HF dataset.


Basically, beaker dataset gets continuously synced at every 3 mins, but it may be difficult to make sure the dataset is synced. As a result, we can prob only queue up an eval job post hoc, not in the middle of the experiment. See https://allenai.slack.com/archives/C6MN19S05/p1724183018709369?thread_ts=1724182496.315879&cid=C6MN19S05